### PR TITLE
Swap new cqc url - introduce baseline changes for old script and new script to deliver data successfully

### PR DESCRIPTION
**Pull request by luke-a-williams**
# Description
I started by swapping things over to the new url, but have reverted those changes in favour of building the new functionality alongside the existing, working, api requests to the old url.
this is because the switch is causing job failures due to differences in schemas and data types, which need to be addressed.
It also gives a chance to resolve technical debt in this area.

Relevant Trello tickets:
Current:
https://trello.com/c/4WXWWnF8/131-update-cqc-api-scripts-to-account-for-move-to-new-url-in-march-2024-must

New tickets:
https://trello.com/c/MBcw87hG/326-proposed-changes-needed-for-switch-the-new-cqc-api
https://trello.com/c/MBcw87hG/326-proposed-changes-needed-for-switch-the-new-cqc-api

The summary of my changes to kick start this process are:
- duplicate the original cqc_api, bulk jobs and test files.
- Have the old jobs point at the old schema and old url, whilst the new job points at the new url and a new schema which needs to be completed in another ticket.
- - The new schemas only differ by the longitude and latitude, the first instance of where an error was occurring, just to get past that step.
- Point the new jobs to a new dataset location so that it doesn't interfere with the old one.
- The new tests have had basic updates to accommodate for the new cqc_api changes, but could still do with more time spent in another ticket refactoring.
- The new api script introduces header information necessary for successful use of the new api.

# How to test
Ensure that both the old and new jobs work.

# Developer checklist
- [x] Unit test
- [x] Linked to Trello ticket
- [x] Documentation up to date

### CODE REVIEW

File: tests/integration/test_cqc_api_new_integration.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import re
+
+EXAMPLE_LOCATION_ID = "1-10000792582"
+LOCATION_ID_REGEX = r"[0-9]-[0-9]{11}"
+
+
+class TestCQCLocationAPIIntegration(unittest.TestCase):
+    def setUp(self):
+        self.PARTNER_CODE_STUB = "PARTNERCODE"
+        pass
```
Comment (james-francis-MT) [inline tests/integration/test_cqc_api_new_integration.py:12]: No need for pass statement now it is doing something

File: utils/cqc_api_new.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+CQC_API_BASE_URL = "https://api.service.cqc.org.uk"
+USER_AGENT = "SkillsForCare"
+
+
+@sleep_and_retry
+@limits(calls=RATE_LIMIT, period=ONE_MINUTE)
+def call_api(url, query_params=None, headers_dict=None):
+    response = requests.get(url, query_params, headers=headers_dict)
+
+    while response.status_code == 429:
```
[the remark this answers is not in the document]
Comment (luke-a-williams) [reply utils/cqc_api_new.py:22]: Great questions, I'm not actually sure but I can write to them and ask so we can get a concrete answer

File: tests/unit/test_cqc_api_new.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+from utils import cqc_api_new as cqc
+
+
+class TestCQCLocationAPI(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
```
[the remark this answers is not in the document]
Comment (luke-a-williams) [reply tests/unit/test_cqc_api_new.py:12]: This was just copying the old test script, and you obviously make a very good point. I'm happy to do some basic refactoring on the new tests if that is desired?

File: tests/integration/test_cqc_api_new_integration.py (hunk 1 above)
Comment (luke-a-williams) [reply tests/integration/test_cqc_api_new_integration.py:12]: Oh good spot I forgot to take that out, thanks!

File: utils/cqc_api_new.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (luke-a-williams) [reply utils/cqc_api_new.py:22]: I've cc'd you in on an email I just sent

File: utils/cqc_api_new.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        response = requests.get(url, query_params, headers=headers_dict)
+
+    if (response.status_code == 403) & (headers_dict is None):
+        raise Exception(
+            "API response: {}, ensure you have set a User-Agent header".format(
+                response.status_code
+            )
+        )
+
+    if response.status_code not in [200, 429]:
```
[the remark this answers is not in the document]
Comment (luke-a-williams) [reply utils/cqc_api_new.py:34]: Good spot, that should be a 403 not a 429, thanks!

**Pull request merged**